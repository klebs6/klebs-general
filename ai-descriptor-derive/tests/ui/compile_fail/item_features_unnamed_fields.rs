// ---------------- [ File: ai-descriptor-derive/tests/ui/compile_fail/item_features_unnamed_fields.rs ]
#![allow(unused_imports)]
use ai_descriptor_derive::*;
use ai_descriptor_trait::*;
use std::borrow::Cow;

#[derive(ItemWithFeatures)]
#[ai("Potion")]
struct Potion(PotionEffect, Duration); // Unnamed fields

struct PotionEffect;

impl ItemFeature for PotionEffect {
    fn text(&self) -> Cow<'_, str> {
        Cow::Borrowed("Gives you super strength.")
    }
}

struct Duration;

impl ItemFeature for Duration {
    fn text(&self) -> Cow<'_, str> {
        Cow::Borrowed("Lasts for 5 minutes.")
    }
}

fn main() {}
