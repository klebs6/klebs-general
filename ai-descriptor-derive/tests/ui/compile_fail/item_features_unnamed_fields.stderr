error: ItemWithFeatures can only be derived for structs with named fields
 --> tests/ui/compile_fail/item_features_unnamed_fields.rs:9:8
  |
9 | struct Potion(PotionEffect, Duration); // Unnamed fields
  |        ^^^^^^
